techniques/applications/systemUpdateCampaign/1.0/systemUpdateCampaign.st

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2022 Normation SAS
+
+bundle agent rudder_system_update_campaign_&RudderUniqueID& {
+  vars:
+      "enabled"       string => "&ENABLE_UPDATE&";
+      "campaign_name" string => "&CAMPAIGN_NAME&";
+      "reboot"        string => "&REBOOT&";
+      "start"         string => "&START&";
+      "end"           string => "&END&";
+      "campaign_id"   string => canonify("${campaign_name}");
+      "status_prefix" string => "rudder_system_update_campaign";
+      "reboot_opt"    string => "";
+
+    pass1::
+      "reboot_opt"    string => "--reboot",
+                          if => "reboot";
+
+    pass2::
+      # We can't pass JSON content as module output as if can exceed the size limit
+      # So we read it from file instead.
+      "report"        string => readfile("${rudder_system_update_module.system_update_report}", "0"),
+                          if => "${status_prefix}_report";
+
+  classes:
+      "has_py"    or => { returnszero("command -v python", "useshell"),
+                          returnszero("command -v python2", "useshell"),
+                          returnszero("command -v python3", "useshell") };
+      "enabled"   expression => strcmp("${enabled}", "enabled");
+      "reboot"    expression => strcmp("${reboot}", "enabled");
+
+      "pass3"     expression => "pass2";
+      "pass2"     expression => "pass1";
+      "pass1"     expression => "any";
+
+    pass2::
+      "call"      expression => usemodule("system_update.py", "--package_manager=${package_module_knowledge.platform_default} --campaign_id=${campaign_id} --start=${start} --end=${end} --node_id=${g.uuid} --agent_schedule=${g.agent_run_interval} ${reboot_opt}"),
+                          if => "enabled.has_py";
+
+  methods:
+    pass1::
+      # Copy module script with ncf helper
+      "module"     usebundle => additional_modules("${this.promise_dirname}/modules/");
+      #
+      # Module returns:
+      #
+      # * A `${status_prefix}_report` class if a report is done
```

Comment:
Some very simple tests in bash to test that the output is well formed could be great